src/firefly/java/edu/caltech/ipac/firefly/server/query/IpacTablePartProcessor.java

```diff
@@ -81,36 +102,19 @@ public boolean isSecurityAware() {
     }
 
     public void downloadFile(URL url, File outFile) throws IOException, EndUserException {
-        URLConnection conn = null;
         try {
             HttpServiceInput inputs = HttpServiceInput.createWithCredential(url.toString());
-            conn = URLDownload.makeConnection(url, inputs.getCookies(), inputs.getHeaders(), false);
-            conn.setRequestProperty("Accept", "*/*");
-            URLDownload.getDataToFile(conn, outFile);
-
-        } catch (MalformedURLException e) {
-            LOGGER.error(e, "Bad URL");
-            throw makeException(e, "Query Failed - bad url.");
-
+            Map<String,String> headers= new HashMap<>();
+            if (inputs.getHeaders()!=null) headers.putAll(inputs.getHeaders());
+            headers.put("Accept", "*/*");
+            URLDownload.getDataToFile(url, outFile,null, headers);
         } catch (FailedRequestException e) {
-            LOGGER.error(e, e.toString());
-            if (conn instanceof HttpURLConnection) {
-                HttpURLConnection httpConn = (HttpURLConnection) conn;
-                int respCode = httpConn.getResponseCode();
-                String desc = respCode == 200 ? e.getMessage() : HttpStatus.getStatusText(respCode);
+            if (e.getResponseCode() > -1) {
+                String desc = e.getResponseCode() == 200 ? e.getMessage() : HttpStatus.getStatusText(e.getResponseCode());
```

Comment:
This logic should be in FailedRequestException.  It should also check for 2xx and not just 200.

Reply:
I was trying not to change the logic of what it was before, but as I look at it this 2xx case will never happen in the exception.